subprojects/platform-jvm/src/main/java/org/gradle/jvm/toolchain/JavaCompiler.java

```diff
@@ -26,4 +28,17 @@
 @Incubating
 public interface JavaCompiler {
 
+    /**
+     * Returns metadata information about this tool
+     *
+     * @return the tool metadata
+     */
+    JavaToolMetadata getMetadata();
```

Comment:
Would it make sense to call it `JavaInstallationToolMetadata`/`JavaToolchainToolMetadata` to make the distinction clear between tools like `java` and `jlink` and the specific `java` tool?

Reply:
That's a good point. Why not `JavaInstallationMetadata`?
There is the potential clash with current `JavaInstallation` ... but that's going away.